packages/angular-cli/commands/serve.ts

```diff
@@ -125,11 +129,6 @@ const ServeCommand = Command.extend({
     return getPort({ port: commandOptions.port, host: commandOptions.host })
       .then((foundPort: number) => {
 
-        if (commandOptions.port !== foundPort && commandOptions.port !== 0) {
```

Comment:
Wow. You are totally right. I completely missed that use case. I'll try to fix it.